src/EditorFeatures/TestUtilities/Workspaces/TestHostDocument.cs

```diff
@@ -238,7 +218,36 @@ public IWpfTextView GetTextView()
 
         public ITextBuffer GetTextBuffer()
         {
-            return this.TextBuffer;
+            var workspace = (TestWorkspace?)_languageServiceProvider?.WorkspaceServices.Workspace;
+
+            if (_textBuffer == null)
+            {
+                Contract.ThrowIfNull(_languageServiceProvider, "To get a text buffer for a TextHostDocument, it must have been parented in a project.");
```

Comment:
```suggestion
                Contract.ThrowIfNull(_languageServiceProvider, "To get a text buffer for a TestHostDocument, it must have been parented in a project.");
```

Reply:
Even better: replacing it with a nameof.